enhancements/network/baremetal-networking.md

```diff
@@ -0,0 +1,260 @@
+authors:
+  - "@jcpowermac"
+reviewers:
+  - "@abhinavdahiya"
+approvers:
+  - "@abhinavdahiya"
+creation-date: 2019-12-10
+last-updated:  2019-12-12
+status: implemented
+see-also:
+  - "https://github.com/openshift/installer/blob/master/docs/design/baremetal/networking-infrastructure.md"
+  - "https://github.com/openshift/installer/blob/master/docs/design/openstack/networking-infrastructure.md"
+  - "https://github.com/openshift/enhancements/pull/61"
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# In Cluster Network Infrastructure
+or internal network services
+or internal networking infrastructure
+or non-cloud network{,ing} {services,infrastructure}
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Many customers still have on-premise infrastructure which includes bare metal, VMware vSphere,
+OpenStack and Red Hat RHV.  These customers would like to use the IPI installation approach while utilizing
+their existing environments. This enhancement would provide networking infrastructure including DNS
+and load balancing required for OpenShift to compute environments that do not provide such services.
+
+## Motivation
+
+### Goals
+
+Install an IPI OpenShift cluster on various on-premise non-cloud platforms that
+provides internal DNS and load balancing that is minimally required for OpenShift
+cluster to run properly.
+
+### Non-Goals
+
+## Proposal
+
+The basis and significant portions for this proposal was taken from existing
+[documentation](https://github.com/openshift/installer/blob/master/docs/design/baremetal/networking-infrastructure.md).
+
+In cluster network infrastructure automates a number
+of components that are handled on other platforms by cloud infrastructure services.
+
+### Load-balanced control plane access
+
+Access to the Kubernetes API (port 6443) from clients both external
+and internal to the cluster should be load-balanced across control
+plane machines.
+
+Access to Ignition configs (port 22623) from clients within the
+cluster should also be load-balanced across control plane machines.
+
+In both cases, the installation process expects these ports to be
+reachable on the bootstrap instance at first and then later on the
+newly-deployed control plane machines.
+
+#### API VIP (Virtual IP)
+
+In cluster network infrastructure, a VIP (Virtual IP) is used to provide
+failover of the API server across the control plane machines
+(including the bootstrap instance). This "API VIP" is provided by the user
+as an `install-config.yaml` [parameter](https://github.com/openshift/installer/blob/master/pkg/types/baremetal/platform.go#L57-L63)
+and the installation process configures `keepalived` to manage this VIP.
+
```

Comment:
would like to see included some restrictions of the API VIP...
example, is this global addressable? or only in the private network? is it part of the private network of cluster or some other private network out side cluster's etc.

Reply:
Seems since we only have one API VIP whether or not it's RFC1918 or not matters less than the requirement that the VIP is accessible both by the cluster and external clients.